### PR TITLE
MNTOR-2934/set UPLOAD_SENTRY_SOURCEMAP correctly

### PR DESCRIPTION
**Pull request by rhelmer**
# References: 
Jira: MNTOR-2934

# Description

When tags are applied, upload sourcemaps to Sentry using GH Action.

# How to test

We want this only to run for releases, we can verify this in the "Source Maps" settings of Sentry's web UI. 

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -9,8 +9,10 @@ import { withSentryConfig } from "@sentry/nextjs";
 const nextConfig = {
   productionBrowserSourceMaps: true,
   sentry: {
-    disableServerWebpackPlugin: process.env.UPLOAD_SENTRY_SOURCEMAPS ?? "false",
-    disableClientWebpackPlugin: process.env.UPLOAD_SENTRY_SOURCEMAPS ?? "false",
+    disableServerWebpackPlugin:
+      process.env.UPLOAD_SENTRY_SOURCEMAPS === "true" ? false : true,
```
Comment (pdehaan) [inline next.config.js:13]: Guessing we wouldn't want something like `process.env.UPLOAD_SENTRY_SOURCEMAPS !== "true"` and avoiding the ternery.

<s>And probably a temp variable is overkill too if you wanted
```js
const uploadSentrySourcemaps = process.env.UPLOAD_SENTRY_SOURCEMAPS === "true";

disableServerWebpackPlugin: !uploadSentrySourcemaps,
disableClientWebpackPlugin: !uploadSentrySourcemaps,
```
</s>

Comment (pdehaan) [reply next.config.js:13]: PS: let me know if you want me to quickly R+ this if you want to get it onto Stage/Prod right meow and just want a blind 👍 to appease the CI Gods.

Comment (rhelmer) [reply next.config.js:13]: Good idea, thanks! I figured out a way to test this before landing (I could test some it locally, but I want to make sure it's working when run from CircleCI), the feedback cycle is so long if we have to push to stage to test: https://github.com/mozilla/blurts-server/pull/4179

Comment (rhelmer) [reply next.config.js:13]: (so no rush to stage but I appreciate the offer!)

Comment (pdehaan) [reply next.config.js]: No problem. Let the record show I have poor ethics and am willing to blind R+ at 9PM for the sake of :yolo: based development (and the LOLs). 🤷

**Comment by rhelmer** [comment]
I can tell from https://github.com/mozilla/blurts-server/pull/4179 this won't work as-is, and we shouldn't pass secrets this way either (`ARG`s get preserved in the docker layer that gets pushed). [ Docker build secrets](https://docs.docker.com/build/building/secrets/) are pretty difficult to set up in this environment as well.

I think it might be easier to upload the symbols from the host environment, which has the sentry auth token set in a CircleCI environment variable. It looks like Sentry can be configured to do that.

**Comment by rhelmer** [comment]
> I can tell from #4179 this won't work as-is, and we shouldn't pass secrets this way either (`ARG`s get preserved in the docker layer that gets pushed). [ Docker build secrets](https://docs.docker.com/build/building/secrets/) are pretty difficult to set up in this environment as well.
> 
> I think it might be easier to upload the symbols from the host environment, which has the sentry auth token set in a CircleCI environment variable. It looks like Sentry can be configured to do that.

I've researched this and while it's definitely possible, it requires more changes (e.g. `Sentry.init` needs to have the release passed to it) - I'd like to let Sentry handle this as part of the build with its webpack plugin if at all possible. I think we might be able to use the GitHub Action runner to do this, although I don't know if there's any potential issues if the sourcemaps come from a different build than what was actually shipped - it seems to work fine in my testing but it feels wrong to be doing multiple builds like this.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -18,3 +18,13 @@ jobs:
       - run: cp .env-dist .env
       # Verify that the build (incl. type-checking) succeeds
       - run: npm run build
+        env:
+          UPLOAD_SENTRY_SOURCEMAPS: false
+        if: ${{ !startsWith(github.ref, 'refs/tags/') }}
```
Comment (pdehaan) [inline .github/workflows/build.yaml:23]: Hypothetically speaking, if I didn't know what exactly this was doing/checking... what is this doing/checking?
Only uploading sourcemaps in stage/prod but not local/Heroku?

File: next.config.js [hunk 3]
```diff
@@ -190,7 +190,7 @@ const sentryWebpackPluginOptions = {
 
   org: "mozilla",
   project: "firefox-monitor",
-  silent: true, // Suppresses all logs
+  silent: false, // Suppresses all logs
```
Comment (pdehaan) [inline next.config.js:193]: Also, why this change? Was this related to sourcemaps, or was for debugging?

File: .github/workflows/build.yaml (hunk 2 above)
Comment (rhelmer) [reply .github/workflows/build.yaml:23]: Yes that's right... I'll add a comment.

File: next.config.js (hunk 3 above)
Comment (rhelmer) [reply next.config.js:193]: I think it makes sense to leave this on, it only runs when tags are applied (or if `UPLOAD_SENTRY_SOURCEMAPS` is true for some other reason). It's a lot more work to verify if it's working otherwise.

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -18,3 +18,15 @@ jobs:
       - run: cp .env-dist .env
       # Verify that the build (incl. type-checking) succeeds
       - run: npm run build
+        env:
+          UPLOAD_SENTRY_SOURCEMAPS: false
+        # Do not upload sourcemaps for non-release (tag) builds.
+        if: ${{ !startsWith(github.ref, 'refs/tags/') }}
```
Comment (mansaj) [inline .github/workflows/build.yaml:24]: I think it's possible that this approach only works for `on: release`

Ref: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

**Review by mansaj — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 4 above)
Comment (rhelmer) [reply .github/workflows/build.yaml:24]: Hm, I think we can either do that, or use `push`. I'd like to avoid duplicating all the steps if possible, I think https://docs.github.com/en/actions/using-workflows/reusing-workflows might be the way to do this?
